app/views/play/cinematic/Command/commands.js

```diff
@@ -40,3 +41,121 @@ export class Sleep extends AbstractCommand {
     return sleep(this.ms)
   }
 }
+
+/**
+ * AnimeCommand is used to turn animejs animation tweens into commands that the Command Runner can play and cancel.
+ */
+export class AnimeCommand extends AbstractCommand {
+  /**
+   * @param {anime} animation The animation that will be run.
+   */
+  constructor (animation) {
+    super()
+    this.animation = animation
+  }
+
+  /**
+   * Starts the animation, returning a cancellable promise that resolves when
+   * animation completes.
+   */
+  run () {
+    return new Promise((resolve, reject) => {
+      this.animation.play()
+      this.animation.complete = resolve
+    })
+  }
+
+  /**
+   * Cancel method ignores the promise and simply moves the animation
+   * to the end.
+   */
+  cancel (promise) {
+    const animation = this.animation
+    animation.seek(animation.duration)
+    return promise
+  }
+}
+
+/**
+ * SyncFunction runs a synchronous function.
+ */
+export class SyncFunction extends AbstractCommand {
+  /**
+   * @param {Function} runFn Synchronous function that doesn't return anything.
+   */
+  constructor (runFn) {
+    super()
+    this.run = () => {
+      runFn()
+      return Promise.resolve()
+    }
+    this.cancel = () => {}
+  }
+}
+
+export class Noop extends AbstractCommand {
+  run () {
+    return Promise.resolve()
+  }
+}
+
+/**
+ * Higher order command that operates on other commands.
+ * Takes in a list of commands and operates on them sequentially.
+ *
+ * Cancelling this command sequentially cancels all included commands.
+ *
+ * Usage traditionally is for adding delays. I.e. Sleep command and then another command.
+ */
+export class SequentialCommands extends AbstractCommand {
+  /**
+   * @param {AbstractCommand[]} commands - List of commands to run sequentially.
+   */
+  constructor (commands) {
+    super()
+    this.commandRunner = new CommandRunner(commands)
+  }
+
+  [run] () {
```

Comment:
I am curious what does it mean to write function name in square brackets `[]`

Reply:
Not sure of the exact name. But it works out the expression for the method.
[Small demo](https://repl.it/@AndrewJakubowic/Example-of-class-methods)

Reply:
You can do a similar thing with object literals.

```
const name = 'Andrew'

const person = {
 [name]: 'blah'
}

// => { Andrew: 'blah }
```

Reply:
Nice, thanks for the examples.

Reply:
Added this to the design slab: https://codecombat.slab.com/posts/cinematic-system-design-2cg8x7si#internal-details-abstract-command